crypto/s2n_fips.h

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+int is_in_fips_mode();
```

Comment:
No, I will change this to "s2n_is_in_fips_mode()".

It is a run-time configuration. FIPS mode requires that libcrypto is build with the FIPS object module. The [FIPS_mode_set()](https://wiki.openssl.org/index.php/FIPS_mode_set()) API is then used to enter FIPS mode.